feat(PocketIC): new builder function PocketIcBuilder::with_server_binary (#4054)
dfinity/ic: packages/pocket-ic/CHANGELOG.md

  (round execution must be triggered separarely, e.g., on a "live" instance or by separate PocketIC library calls).
- The function `PocketIc::set_certified_time` to set the current certified time on all subnets of the PocketIC instance.
- The function `PocketIc::update_call_with_effective_principal` is made public. It is helpful, e.g., for
  modeling management canister calls that need to be routed to the right subnet using effective principals.

> - The function `PocketIcBuilder::with_server_binary` to provide a path to the PocketIC server binary used to start a server process if necessary.

Author: ```suggestion
- The function `PocketIcBuilder::with_server_binary` to provide a path to the PocketIC server binary used instead of the environment variable `POCKET_IC_BIN`.
```

Author: > Was there not a third possible source? Having a binary present next to the cargo.toml file?

Indeed, this is used if users do not specify anything: here I wanted to point out how you can override the default by using the new function taking precedence over POCKET_IC_BIN. I'm not sure the changelog is the right place for documentation. We should include them in the HOWTO.md if at all. Do you feel strongly about that?

Author: I've added all the three ways into the panic message printed in case all the three ways fail.

Reviewer: >I'm not sure the changelog is the right place for documentation. We should include them in the HOWTO.md if at all.

Agreed. Yes, I would put a note in the docstring of the new builder method _and_ a small section in the HOWTO.

Author: > I would put a note in the docstring of the new builder method

mentioning all the alternatives in the docstring by now

> and a small section in the HOWTO

I realized we document the options in the panic message printed if no path to the server binary could be determined by using one of the three options. Is this not enough after all?

Reviewer: alright
